dnsmasq: Rewrite docs page for new version including DHCP (#678)
opnsense/docs: source/manual/dnsmasq.rst

==================
Dnsmasq DNS & DHCP
==================

.. contents:: Index

`Dnsmasq` is a lightweight and easy to configure DNS forwarder and DHCPv4/DHCPv6 server.

> It is considered the replacement for `ISC-DHCP` in small and medium sized setups,

Reviewer: ```suggestion
It is considered the replacement for `ISC-DHCP` in small and medium sized setups
```

In the configuration examples further below, we will always combine `Unbound` with `Dnsmasq`.

DHCP Service
-----------------------------

`Dnsmasq` is the perfect DHCP server for small and medium sized setups. The configuration is straight forward, and since it can register the DNS names of leases,
it can replicate the simplicity known from consumer routers.

If HA for DHCP is a requirement, split pools can be configured for two `Dnsmasq` instances. With a dhcp reply delay, the secondary instance will only answer when
the first instance is unresponsive. DHCPv6 and Router Advertisements are also an option for small HA setups that do not have fast failover requirements,
as IPv6 failover can take up to 30 seconds with available configuration options.

For larger enterprise setups, `KEA DHCP` can be a viable alternative. It supports lease synchronisation via REST API, which means both DHCP servers keep track
of all existing leases and do not need split pools. It is also far more scalable if there are thousands of leases.

The tradeoff using `KEA DHCP` is a more complicated setup, especially when custom DHCP options are needed. DNS registration is also not possible.

> With this in mind, pick the right choice for your setup. When in doubt, using `Dnsmasq` can be the best choice.

Reviewer: ```suggestion
With this in mind, pick the right choice for your setup. When in doubt, our advise is to use `Dnsmasq` .
```
